### PR TITLE
basicsetup

### PR DESCRIPTION
**Pull request by ngyihkai**
Ready

### CODE REVIEW

File: app/controllers/registers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
+class RegistersController < ApplicationController
+  before_action :logged_in_user
+
+  def new
+    @register = Register.new
+  end
+
+  def create
+    @user = User.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/registers_controller.rb:9]: User should be current_user, right?
=> recommend the handler from line 9-> 19:
```
@course = Course.find_by(id: params[:course_id])
@register = Register.new(register_params)

if @register.save
  flash[:info] = "Waiting approval"
else
  flash[:danger] = "Register failed"
end

redirect_back(fallback_location: @current_user)
```

File: app/controllers/registers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      redirect_back(fallback_location: @user)
+    end
+  end
+
+  def edit
+    @register = Register.find(params[:id])
+  end
+
+  def update
+    @course = Course.find_by(params[:id])
```
Comment (huongnt-2545) [inline app/controllers/registers_controller.rb:27]:
```suggestion
    @course = Course.find_by(id: params[:course_id])
```

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -1,77 +1,54 @@
...
+
+    unless @user
+      flash[:danger] = t(:user_not_found)
+      redirect_to home_path
+    end
+  end
 
   def new
-    @user = User.new
+     @user = User.new
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:22]: wrong indent

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -1,77 +1,54 @@
...
-      redirect_to @user
+      flash[:info] = t(:check)
+      redirect_to root_url
     else
       render :new
     end
   end
 
   def update
-    @user = User.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:29]: please keep this handler, don't change

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,77 +1,54 @@
...
-    if @user.update_attributes(user_params)
-      flash[:success] = "Profile updated"
-      redirect_to @user
-    else
-      render "edit"
-    end
+    @user = User.find(params[:id])
   end
 
-  def destroy
```
Comment (huongnt-2545) [inline app/controllers/users_controller.rb:38]: keep this, why do you remove it?

**Comment by ngyihkai** [comment]
Ready

File: app/controllers/sessions_controller.rb [hunk 6]
```diff
@@ -21,7 +21,12 @@ def create
   end
 
   def destroy
-    log_out if logged_in?
-    redirect_to home_path
+  user = User.find_by(id: params[:id])
```
Comment (huongnt-2545) [inline app/controllers/sessions_controller.rb:24]: wring indent

File: app/controllers/sessions_controller.rb [hunk 7]
```diff
@@ -21,7 +21,12 @@ def create
...
+    if user&.destroy
+      flash[:success] = "User deleted"
+    else
+      flash[:danger] = "Delete fail!"
+    end
+  redirect_to users_url
```
Comment (huongnt-2545) [inline app/controllers/sessions_controller.rb:30]: wrong indent

File: test/controllers/courses_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/controllers/courses_controller_test.rb:1]: delete this file

File: test/fixtures/courses.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/courses.yml:1]: delete this file

File: test/fixtures/registers.yml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# Read about fixtures at https://api.rubyonrails.org/classes/ActiveRecord/FixtureSet.html
```
Comment (huongnt-2545) [inline test/fixtures/registers.yml:1]: delete this file

File: test/models/course_test.rb [hunk 11]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/models/course_test.rb:1]: delete this file

File: test/models/register_test.rb [hunk 12]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (huongnt-2545) [inline test/models/register_test.rb:1]: delete this file

**Review by huongnt-2545 — APPROVED** [review]
merged

**Pull request merged**